compiler/resolution/functionResolution.cpp

```diff
@@ -4985,7 +4985,13 @@ static void resolveNew(CallExpr* call) {
             if (isBlockStmt(call->parentExpr) == true) {
               call->insertBefore(def);
             } else {
-              call->parentExpr->insertBefore(def);
+              Expr* parent = call->parentExpr;
```

Comment:
(off list, Mike said to merge for now, but we might want to take a look at this function again soon)